components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/utils/mappings/APIMappingUtil.java

```diff
@@ -228,6 +227,67 @@ public static APIDTO fromAPItoDTO(org.wso2.carbon.apimgt.api.model.API model) th
 
     }
 
+    private static String mapVisibilityFromDTOtoAPI(APIDTO.VisibilityEnum visibility) {
+        switch (visibility) {
+            case PUBLIC:
+                return APIConstants.API_GLOBAL_VISIBILITY;
+            case PRIVATE:
+                return APIConstants.API_PRIVATE_VISIBILITY;
+            case RESTRICTED:
+                return APIConstants.API_RESTRICTED_VISIBILITY;
+            case CONTROLLED:
+                return APIConstants.API_CONTROLLED_VISIBILITY;
+            default:
+                return null; // how to handle this?
+        }
+    }
+    private static APIDTO.VisibilityEnum mapVisibilityFromAPItoDTO(String visibility) {
+        switch (visibility) { //public, private,controlled, restricted
```

Comment:
Please consider that one method is mapping DTO to backend model while the other mapping vice versa.
Method comments were added in PR #1091